src/Routing/Router.php

```diff
@@ -559,6 +559,7 @@ public static function url($url = null, $full = false)
 
         $request = static::getRequest(true);
         if ($request) {
+            $request->params += $params;
```

Comment:
Oh yeah, I didn't recognized $request is an object. Was thinking about an array.
So maybe rather:

    $tmp = $request->params;
    $tmp += $params;
    $params = $tmp

?

In line 613 all the necessary indices have to set. By just set $params with $request->params there is a possibility that indices aren't set. It overwrites your defaut indices of line 552.